vars/cloudFoundryDeploy.groovy

```diff
@@ -276,9 +278,28 @@ def deployMta (config) {
     deploy(apiStatement, deployStatement, config, null)
 }
 
+private checkAndUpdateDeployTypeForNotSupportedManifest(Map config){
+    String manifestFile = config.cloudFoundry.manifest ?: 'manifest.yml'
+    if(config.deployType == 'blue-green' && fileExists(manifestFile)){
+        Map manifest = readYaml file: manifestFile
+        List applications = manifest.applications
+        if(applications) {
+            if(applications.size()>1){
+                error "[${STEP_NAME}] Your manifest contains more than one application. For blue green deployments your manifest file may contain only one application."
+            }
+            if(applications.size==1 && applications[0]['no-route']){
```

Comment:
could it be possible that the user just does not define a route?
Then you could add `!applications[0].routes`

Reply:
``` 
If a domain or hostname is not specified, then a route will be created using the app name and the default shared domain (see Shared Domains). The following command pushes the app myapp, creating the route myapp.shared-domain.example.com from the default shared domain shared-domain.example.com. If the route has not already been created in another space this command also maps it to the app.
``` 
So it always will create a route.

Reply:
thx for clarification